src/utils/react-query-factory.js

```diff
@@ -0,0 +1,93 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```

Comment:
[nit/suggestion] Rename file to `queryKeyFactory.js` or similar (no strong feelings on hyphens vs. camelCase), given react is largely implied and I imagine a common search term to try to find this file would be related to query _key_ so might make sense to include "key" in the file name too?